financial-templates-lib/clients/ExpiringMultiPartyClient.js

```diff
@@ -90,32 +93,49 @@ class ExpiringMultiPartyClient {
     return this.lastUpdateTimestamp;
   }
 
-  async update() {
-    // Since this function can have a relatively long run-time, we should be safe and check contract state explicitly at the same block height.
-    // This way we won't experience errors where the contract state changes between calls.
-    this.latestBlock = await this.web3.eth.getBlockNumber();
+  // Returns web3.eth.Contract with an ERC20 token interface at
+  // the tokenCurrency address
+  async getTokenCurrency() {
+    const tokenCurrencyAddress = await this.emp.methods.tokenCurrency().call({}, this.latestBlock);
+    const tokenCurrency = new this.web3.eth.Contract(Token.abi, tokenCurrencyAddress);
 
-    this.collateralRequirement = this.toBN(
-      (await this.emp.methods.collateralRequirement().call(undefined, this.latestBlock)).toString()
-    );
-    this.liquidationLiveness = Number(await this.emp.methods.liquidationLiveness().call(undefined, this.latestBlock));
+    // Thank you inconsistent web3
```

Comment:
How could it be possible that `tokenCurrency.address === undefined/null`?

Reply:
No clue :eyes:

Reply:
If that call returns `null` or `undefined` then something if wrong with the `emp` instance. Either you've provided an address that does not exist on the chain or some other reason why it returned this. 

In either case, there are no other situations where we return the full `web3.eth.Contract` objects like this from EMPClients.

Having the Token like this in the EMP Client is muddling the point of this client.

Reply:
Agreed with both of Chris' points, you shouldn't have to check for `null`/`undefined` and this client is meant to be artifact-less

Reply:
Agreed with Chris' points:
- We should not have to check that `tokenCurrency.address` is `null`, this is an assumption we make
- The Token contract shouldn't have to be returned from this class. I see you ultimately use it in the Liquidator class which already constructs it in liquidator/index.js